syntactic/src/main/scala/syntactic/Checker.scala

```diff
@@ -11,20 +11,20 @@ trait Checker {
   /**
    * Check whether the input tree uses only allowed features
    *
-   * @param tree the tree to be checked
+   * @param node the tree to be checked
    * @return Some[Violation] if a violation was found, None o.w.
    */
-  def checkTree(tree: Tree): Option[Violation]
+  def checkNode(node: Tree): Option[Violation]
```

Comment:
Interesting, so both `checkNode` and `checkTree` take a `Tree` as parameter but `checkNode` works only on “that tree”, whereas `checkTree` works on all the descendants of the tree, right?

Reply:
Yes exactly, I think that it is clearer that way.